assets/src/edit-story/components/helpCenter/navigator/bottomNavigation.js

```diff
@@ -54,14 +62,36 @@ export function BottomNavigation({
   isNextDisabled,
   isPrevDisabled,
 }) {
+  // If either the prev or next has focus and become disabled,
+  // we want to force focus to the companion instead of losing
+  // it to the canvas.
+  const prevButtonRef = useRef(null);
+  const nextButtonRef = useRef(null);
+  const [local, setLocal] = useState({
+    isPrevDisabled,
+    isNextDisabled,
+  });
+  // We want to force focus one render before we disable the actual
+  // DOM elements so we can see if they're currently focused.
+  useEffect(() => {
+    [
+      isPrevDisabled && document.activeElement === prevButtonRef.current,
+      isNextDisabled && document.activeElement === nextButtonRef.current,
+    ].forEach((condition) => onCondition(condition)(forceFocusCompanion));
+    setLocal({ isPrevDisabled, isNextDisabled });
+  }, [isPrevDisabled, isNextDisabled]);
+
```

Comment:
this is really well organized and explained

Reply:
let me know if the curried conditional is a little extra lol. fealt good when I was writing it, then after I pushed it up wondered if I've gone to far

Reply:
The curried conditional works well. It may be more readable to use arguments in one function instead but I don't mind either way.

Tbh it's readable the way you did it.
```js
    ].forEach((condition) => onCondition({ onCondition, fn: forceFocusCompanion }));
```